src/js/strategies/CitrixVDIStrategy.js

```diff
@@ -0,0 +1,131 @@
+                    reject(); })
+            })
+        }
+        window.CitrixWebRTC.initLog(global.connect.getLog());
+        console.log("CitrixVDIStrategy initialized");
+    }
+
+    // the following functions are rtc_peer_connection_factory related functions
+    // check if the browser supports early media connection
+    _isBrowserSupported(){
+        // Citrix WebRTC SDK doesn't support early media connection
+        return false;
+    }
+
+    _createRtcPeerConnection(rtcPeerConnectionConfig, rtcPeerConnectionOptionalConfig) {
+        return new window.CitrixWebRTC.CitrixPeerConnection(rtcPeerConnectionConfig, rtcPeerConnectionOptionalConfig);
+    }
+
+    // the following functions are rtc_session related functions
+    _gUM(constraints) {
+        return window.CitrixWebRTC.getUserMedia(constraints);
+    }
+
+    createOfferStateOnEnter(_pc, stream) {
+        stream.getTracks().forEach(track => {
+            _pc.addTransceiver(track, {});
+        });
+    }
+
+    acceptStateOnEnter(self, rtcSession) {
+        console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription`);
+
+        const answerSessionDescription = self._createSessionDescription({ type: 'answer', sdp: self._sdp });
+        console.log(`ConnectRTC - rtc_session - AcceptState - answerSessionDescription: `, answerSessionDescription);
+
+        rtcSession._pc.setRemoteDescription(answerSessionDescription, () => {
+            console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription - success`);
+            var remoteCandidatePromises = Promise.all(self._candidates.map(function (candidate) {
+                var remoteCandidate = self._createRemoteCandidate(candidate);
+                self.logger.info('Adding remote candidate', remoteCandidate);
+                console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription - Adding remote candidate - remoteCandidate`, remoteCandidate);
+                return rtcSession._pc.addIceCandidate(remoteCandidate);
+            }));
+            remoteCandidatePromises.catch(reason => {
+                console.log(`ConnectRTC - rtc_session - AcceptState - rtcSession._pc.setRemoteDescription - Adding remote candidate - failed`, reason);
+                self.logger.warn('Error adding remote candidate', reason);
+            });
+            rtcSession._sessionReport.setRemoteDescriptionFailure = false;
+            self._remoteDescriptionSet = true;
+            self._checkAndTransit();
+        }, (error) => {
+            console.error(`ConnectRTC - rtc_session - AcceptState - setRemoteDescription - error:`, error);
+            rtcSession._stopSession();
+            rtcSession._sessionReport.setRemoteDescriptionFailure = true;
+            self.transit(new FailedState(rtcSession, RTC_ERRORS.SET_REMOTE_DESCRIPTION_FAILURE));
+        });
+    }
+
+    onIceStateChange(evt, _pc) {
+        return _pc.iceConnectionState;
```

Comment:
Do we want to log this? or is the method you are calling logging it?

Reply:
Checked with @yiming-amzn, this is not required.